### DISCUSSION TITLE
Get username in system provisioning independent from hostname?

### DISCUSSION THREAD

**Discussion by fwilhe** [lima-vm/lima#2256] [Q&A]
Hi,

related to #1015 I am wondering how I can get the hostname so it does not depend on the name of my local user on my host os.

My use-case is that I want to share a lima yaml file where I'm installing docker in a system provisioning script (so it's all as root), but I want to add my non-root user to the docker group so I can use docker without sudo.

I'd be fine with using an environment variable but I could not find any.

I'd also be fine with a `LIMA_DEFAULT_USERNAME` variable/setting.

So in my setup I have a line like `usermod -a -G docker MYUSERNAME` where the username currently depends on the host os which makes it hard to share/reuse the yaml file. Is there a way to make this generic?

Another reason why people might want to use a username that is not equal to the username of the host os is if the host os username is something that should not be shared. So if I copy/paste my default prompt from my vm it always has the username in it which I don't want to share for privacy reasons. I'd be happy with a generic 'lima' username.

**Comment by afbjorklund** [comment]
The groups were discussed, not sure about user:

* https://github.com/lima-vm/lima/discussions/691
* https://github.com/lima-vm/lima/issues/1686

BTW; I think the generated ssh key also leaks user@host (since it is the default "comment", in the `user.pub`)

**Reply by fwilhe** [reply]
Good point about the ssh key

For me currently it's two separate but related questions (sorry for putting them in one issue)

- How can I generically address my non-root user in the system provisioning script so it will not only work on my laptop?
- How can I avoid leaking my host's username (which I can't freely choose on my work laptop) by sharing screenshots/logs/yaml files

**Reply by afbjorklund** [reply]
I think the user is covered by #1015, but we probably need a new issue for the lima-provided ssh key. There's one about users:

* https://github.com/lima-vm/lima/issues/1617

It should be something simple like providing a default value (such as `-C "lima"`) for the comment, instead of the user@host?

> `ssh-keygen` will by default write keys in an OpenSSH-specific format.  This format is preferred as it offers better protection for keys at rest as well as allowing storage of key comments within the private key file itself. 
> The key comment may be useful to help identify the key.  The comment is initialized to “user@host” when the key is created, but can be changed using the -c option.
